piff/model.py

```diff
@@ -110,6 +110,26 @@ def fit(self, star):
         """
         raise NotImplemented("Derived classes must define the fit function")
 
+    def solve(self, star_list, logger=None):
+        """Solve for any model coefficients given some data.
+
+        :param star_list:   A list of Star instances to fit model parameters with
+        :param logger:      A logger object for logging debug info. [default: None]
+
+        By default does NOTHING since you may have no hyperparameters to put in
+        """
+        pass
```

Comment:
The optics model as implemented won't actually have this either, but I think it's a feature of all models, just like with interpolators: if an interpolator sometimes needs to learn the hyperparameters to go from positions to fit parameters, it seems reasonable to me that a model might need to learn hyperparameters to either go from images to fit parameters or fit parameters to images.

Here is a non-trivial example: PCA of the images. You can't determine the PC's in advance; you have to feed it some set of images for it to make that.

Another example is if we want to interpolate over whitened stars (ie mean subtracted and/or variance normalized) or apply a model to whitened stars. I am not sure why we would want to do this, but in that case you would need to 'solve' for the mean and/or variance of the pixels so that you can properly fit and draw.